Support for Elasticsearch 2.x

akonczak opened the pull request:
Support for ES 2.x - Spring Data ES has dependency with this pull  - DATAES-219

akonczak committed:
Support for Elasticsearch 2.x

Shredder121 commented:
Thanks for the PR.
One idea: use the `com.querydsl.elasticsearch2` package, to prevent clashes with ClassLoaders not liking two classes with the same name and package.

Shredder121 commented on querydsl-elasticsearch2/pom.xml:
```diff
@@ -0,0 +1,151 @@
+    <plugins>
+      <plugin>
+        <groupId>com.springsource.bundlor</groupId>
+        <artifactId>com.springsource.bundlor.maven</artifactId>
```
`<version />` not present.
I notice that it's not there on the original elasticsearch module. Thoughts @timowest ?

Shredder121 replied:
Maybe put it in a parent pom?

Shredder121 commented on querydsl-elasticsearch2/src/test/java/com/querydsl/elasticsearch/PackageVerification.java:
```diff
@@ -0,0 +1,54 @@
+    @Test
+    public void Verify_Package() throws Exception {
+        String version = System.getProperty("version");
+        verify(new File("target/querydsl-elasticsearch-" + version + "-apt-one-jar.jar"));
```
`target/querydsl-elasticsearch2-" + version + "-apt-one-jar.jar`

Shredder121 commented on querydsl-elasticsearch2/pom.xml:
```diff
@@ -0,0 +1,151 @@
+                </property>
+              </systemProperties>
+              <includes>
+                <include>com/mysema/query/PackageVerification.java</include>
```
`com/querydsl/elasticsearch/PackageVerification.java`

But I recommend having this under `com.querydsl.elasticsearch2/PackageVerification.java`

@timowest I believe this is also wrong in the original `querydsl-elasticsearch` project.
Its `javax.annotation.processing.Processor` file is still outdated.

I'll open up a PR.

akonczak commented:
Additional changes: 
-renamed package to not clash with ES 1.x
-added version for bundler (main pom)
-changed paths for PackageVerification in all poms to reflect correct location

Shredder121 commented:
I forgot to change some `.` to `/` in my comment, so that should be `com/querydsl/elasticsearch2/PackageVerification.java`
But I first want to hear @timowest's opinion on some matters.

Anyway, thank you for spearheading this.

kevinleturc commented:
Hi guys, thanks to review this PR on both projects ;).
I'll investigate a bit on spring bundlor plugin, I'm not sure if it's needed.

kevinleturc commented:
Spring bundlor plugin seems to be not used during a classic compilation.
It is also present in dynamodb and hazelcast. Maybe we can remove it ?

akonczak commented:
@Shredder121 

My mistake.
PackageVerification is only available only for ES/ES2, moreover test(s) will fails if you run it.

`Running com.querydsl.elasticsearch.PackageVerification
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.118 sec <<< FAILURE! - in com.querydsl.elasticsearch.PackageVerification
Verify_Package(com.querydsl.elasticsearch.PackageVerification)  Time elapsed: 0.116 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<com.[querydsl].apt.QuerydslAnnotat...> but was:<com.[mysema.query].apt.QuerydslAnnotat...>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.querydsl.elasticsearch.PackageVerification.verify(PackageVerification.java:51)
	at com.querydsl.elasticsearch.PackageVerification.Verify_Package(PackageVerification.java:37)
`

Shredder121 commented:
Yes, `PackageVerification` is a sanity check, a safety net if you will.
It was supposed to run all the time, but it didn't..

Shredder121 commented on querydsl-elasticsearch2/src/apt/META-INF/services/javax.annotation.processing.Processor:
```diff
@@ -0,0 +1 @@
+com.mysema.query.apt.QuerydslAnnotationProcessor
```
This should point to `com.querydsl.apt.QuerydslAnnotationProcessor`.

akonczak commented:
added PackageVerification for all module + basic test to verify that core classes are available.

Shredder121 commented:
@kevinleturc That particular plugin can indeed be removed.

In the `querydsl-root` pom we have configured the `org.apache.felix:maven-bundle-plugin`, so that should actually be used there.
```xml
<plugin>
  <groupId>org.apache.felix</groupId>
  <artifactId>maven-bundle-plugin</artifactId>
</plugin>
```

akonczak commented:
i will do that later today

Shredder121 commented:
That's okay.

akonczak commented:
ok all done.

akonczak commented:
Do we have any date when this pull request can be approved ?
